data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java

```diff
@@ -107,13 +107,25 @@ public HttpRequestBuilder withURL(String url) throws DataClientException {
         return this;
     }
 
-    public HttpRequest build() {
+    public HttpRequest build() throws DataClientException {
+        // If has authorization header, ensure it is not sent over insecure channel
+        boolean hasAuth = request.getHeaders().stream().anyMatch(h -> h.getName().equalsIgnoreCase(HttpHeaderName.AUTHORIZATION.toString()));
+        if (hasAuth) {
+            URL url = request.getUrl();
+            boolean isHttp = url.getProtocol().equalsIgnoreCase("http");
+            boolean isLocalhost = url.getHost().toLowerCase().startsWith(CloudInfo.LOCALHOST);
```

Comment:
this is a bit risky.
localhostmyass.com is a valid target 😅
You'd want to regex match the host against `^localhost(\:[0-9]+)?$`

Reply:
good point (though this is the logic we already have).

I think the host method returns it without the port, so I'll just check equality